src/pages/index.js

```diff
@@ -0,0 +1,359 @@
+    if (!deleteCardModal) {
+      console.error("Delete modal not found!");
+      return;
+    }
+    cardToDelete = cardElement;
+    openModal(deleteCardModal);
+  }
+
+  if (confirmDeleteButton) {
+    confirmDeleteButton.addEventListener("click", () => {
+      if (!cardToDelete) return;
+
+      const deleteButton = confirmDeleteButton;
+      const originalText = deleteButton.textContent;
+
+      renderLoading(true, deleteButton, originalText);
+
+      api
+        .removeCard(cardToDelete.dataset.cardId)
+        .then(() => {
+          cardToDelete.remove();
+          closeModal(deleteCardModal);
+        })
+        .catch((error) => showErrorMessage(error, "Deleting card"))
+        .finally(() => {
+          renderLoading(false, deleteButton, originalText);
+        });
+    });
+  }
+
+  if (profileEditButton) {
+    profileEditButton.addEventListener("click", () => {
+      resetForm(profileForm, settings);
+      inputProfileName.value = profileName?.textContent.trim() || "";
+      inputProfileDescription.value =
+        profileDescription?.textContent.trim() || "";
+      openModal(profileEditModal);
+    });
+  }
+
+  if (profileForm) {
+    profileForm.addEventListener("submit", handleProfileFormSubmit);
+
+    function handleProfileFormSubmit(evt) {
+      function makeRequest() {
+        return api
+          .updateUserInfo({
+            name: inputProfileName.value.trim(),
+            about: inputProfileDescription.value.trim(),
+          })
+          .then(() => {
+            profileName.textContent = inputProfileName.value.trim();
+            profileDescription.textContent =
+              inputProfileDescription.value.trim();
+            closeModal(profileEditModal);
+          });
+      }
+
+      handleSubmit(makeRequest, evt);
+    }
+  }
+
+  if (profileAvatarEditButton) {
+    profileAvatarEditButton.addEventListener("click", () => {
+      resetForm(avatarForm, settings);
+      openModal(editAvatarModal);
+    });
+  }
+
+  if (avatarForm) {
+    avatarForm.addEventListener("submit", handleAvatarFormSubmit);
+  }
+  function handleAvatarFormSubmit(evt) {
+    function makeRequest() {
+      return api.updateProfileAvatar(inputAvatarUrl.value.trim()).then(() => {
+        profileAvatar.src = inputAvatarUrl.value.trim();
+        closeModal(editAvatarModal);
+      });
+    }
+
+    handleSubmit(makeRequest, evt);
+  }
+
+  if (openAddCardModalButton) {
+    openAddCardModalButton.addEventListener("click", () =>
+      openModal(addCardModal)
+    );
+  }
+
+  if (addCardForm) {
+    addCardForm.addEventListener("submit", handleCardFormSubmit);
+
+    function handleCardFormSubmit(evt) {
+      function makeRequest() {
+        return api
+          .addNewCard({
+            name: inputCardCaption.value.trim(),
+            link: inputCardImageLink.value.trim(),
+          })
+          .then((cardData) => {
+            renderCard(cardData);
+            closeModal(addCardModal);
```

Comment:
* You need to **`disable`** the submit button in the `card form` if the inputs are **empty**. It’s needed because I can click the button and add an empty card, but it should **not** happen. After the first submission, the button gets activated, and you need to **deactivate** it again. Even if I can't add an empty card, the button should look **disabled** (gray) anyway and have a `disabled` attribute.